src/cli/commands/start.ts

```diff
@@ -1,77 +1,208 @@
-import concurrently from "concurrently";
+import chalk from "chalk";
+import { Command } from "commander";
+import concurrently, { CloseEvent } from "concurrently";
 import { CommandInfo } from "concurrently/dist/src/command";
 import fs from "fs";
 import path from "path";
+import { execSync } from "child_process";
 import { DEFAULT_CONFIG } from "../../config";
 import {
+  askNewPort,
+  configureOptions,
   createStartupScriptCommand,
   detectTargetCoreToolsVersion,
   getCoreToolsBinary,
   getNodeMajorVersion,
   isAcceptingTcpConnections,
   isCoreToolsVersionCompatible,
   isHttpUrl,
+  isUserOption,
   logger,
+  matchLoadedConfigName,
+  parseServerTimeout,
+  parsePort,
   parseUrl,
   readWorkflowFile,
 } from "../../core";
-import builder from "../../core/builder";
+import { swaCLIEnv } from "../../core/env";
+import { getCertificate } from "../../core/ssl";
 let packageInfo = require("../../../package.json");
 
-export async function start(startContext: string, options: SWACLIConfig) {
+export default function registerCommand(program: Command) {
+  program
+    .command("start [configName|outputLocation|devServerUrl]")
+    .usage("[configName|outputLocation|devServerUrl] [options]")
+    .description("start the emulator from a directory or bind to a dev server")
+    .option("--app-location <path>", "the folder containing the source code of the front-end application", DEFAULT_CONFIG.appLocation)
+    .option("--api-location <path>", "the folder containing the source code of the API application", DEFAULT_CONFIG.apiLocation)
+    .option("--output-location <path>", "the folder containing the built source of the front-end application", DEFAULT_CONFIG.outputLocation)
+    .option("--dev-server-url <url>", "connect to the dev server at this URL instead of using output location", undefined)
+    .option("--api-server-url <url>", "connect to the api server at this URL instead of using output location", undefined)
+    .option<number>("--api-port <apiPort>", "the API server port passed to `func start`", parsePort, DEFAULT_CONFIG.apiPort)
+    .option("--host <host>", "the host address to use for the CLI dev server", DEFAULT_CONFIG.host)
+    .option<number>("--port <port>", "the port value to use for the CLI dev server", parsePort, DEFAULT_CONFIG.port)
+    .option("--run-build", 'run "swa build" before starting the emulator', false)
+
+    .option("--ssl", "serve the front-end application and API over HTTPS", DEFAULT_CONFIG.ssl)
+    .option("--ssl-cert <sslCertLocation>", "the SSL certificate (.crt) to use when enabling HTTPS", DEFAULT_CONFIG.sslCert)
+    .option("--ssl-key <sslKeyLocation>", "the SSL key (.key) to use when enabling HTTPS", DEFAULT_CONFIG.sslKey)
+    .option("--run <startupScript>", "run a custom shell command or script file at startup", DEFAULT_CONFIG.run)
+    .option<number>(
+      "--server-timeout <time>",
```

Comment:
Is there a reason this option was renamed from `--devserver-timeout` to `--server-timeout`? It'd be nice to keep it `--devserver-timeout` for consistency and reducing breaking changes.

Reply:
@manekinekko @sinedied

Reply:
We were discussing these changes with @sinedied during our review the other day. I think we agreed to rename this flag back to `--devserver-timeout`. We haven't had the time to revert the change.

We are also introducing these new flags:
- `--app-devserver-url`
- `--api-devserver-url`

FYI, another breaching change introduced is that the `--server-timeout` flag now accepts values in seconds, **not milliseconds**.

Reply:
@sinedied  can you revert the change in your pending PR (https://github.com/Azure/static-web-apps-cli/pull/472)?

`--server-timeout` --> `--devserver-timeout`

Reply:
> We are also introducing these new flags:
> 
> * --app-devserver-url
> * --api-devserver-url

Will these flags be replacing any existing flags, or are they totally new?